src/Core/functions.php

```diff
@@ -489,3 +492,39 @@ function toBool(mixed $value): ?bool
 
     return null;
 }
+
+/**
+ * Converts a value to a DateTimeInterface.
+ *
+ *  integer  - value is treated as a Unix timestamp
+ *  string - value is treated as a ISO-8601 (Atom) formatted timestamp
+ *  Other values returns as null.
+ *
+ * @param mixed $value The value to convert to DateTimeInterface.
+ * @return \DateTimeInterface|null Returns a DateTimeInterface if parsing is successful, or NULL otherwise.
+ * @since 5.1.0
+ */
+function toDateTime(mixed $value): ?DateTimeInterface
```

Comment:
would a toDate() also be useful here?

Reply:
I can add this function, but it will not be used by anything (yet). I think @markstory and I both feel that casting request parameters to dates/datetimes should live in userland code because of the wide variety of formats that dates can appear as in apps. So I was going to remove date/datetime support from `ServerRequest::asType`.

Reply:
I've added it, but I've also changed the return types for `toDateTime()` and `toDate()` to `\Cake\I18n\DateTime` and `\Cake\I18n\Date` respectively. I hope that was the right choice.

I was unsure about how to test the global functions but then I found `tests/TestCase/Core/FunctionsTest.php` so I need to go write some more tests there.